kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ImageInstanceDescriptor.java

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.container.orchestration;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Object which represents a image. Used to track created images, and images
+ * that exist in the container engine.
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ * @since 2.4
+ *
+ */
+@ProviderType
+public class ImageInstanceDescriptor {
+
+    private String imageName = "";
+    private String imageTag = "";
+    private String imageId = "";
+    private String imageAuthor = "";
+    private String imageArch = "";
+    private long imageSize = 0;
+    private Map<String, String> imageLabels;
```

Comment:
This is called imageLabels but then it is returned with the method imageTags. What is the correct naming?
Furthermore, why the previous fields are initialised as empty and imageLabels is not initialised? Can we align?